src/blockdev.rs

```diff
@@ -809,6 +809,56 @@ pub fn lsblk_single(dev: &Path) -> Result<HashMap<String, String>> {
     Ok(devinfos.remove(0))
 }
 
+fn lsblk_all() -> Result<Vec<HashMap<String, String>>> {
+    let mut cmd = Command::new("lsblk");
+    cmd.arg("--noheadings")
+        .arg("--nodeps")
+        .arg("--list")
+        .arg("--paths")
+        .arg("--output")
+        .arg("NAME");
+    let output = cmd_output(&mut cmd)?;
+    for dev in output.lines() {
+        if let Ok(mut fd) = std::fs::File::open(dev) {
+            let _ = reread_partition_table(&mut fd);
+        }
+    }
+    udev_settle()?;
+    let mut result: Vec<HashMap<String, String>> = Vec::new();
+    for dev in output.lines() {
+        let mut info = lsblk(Path::new(dev), true)?;
+        result.append(&mut info);
+    }
+    Ok(result)
+}
+
+pub fn get_filesystems_with_label(label: &str) -> Result<Vec<String>> {
+    let label = label.to_string();
+    let all = lsblk_all()?;
+    let all = all
+        .iter()
+        .filter(|v| v.get("LABEL") == Some(&label))
+        .collect::<Vec<_>>();
+
+    let mut uuids = HashSet::new();
+    let result = all
+        .iter()
+        .filter(|v| match v.get("UUID") {
+            Some(uuid) => {
+                if !uuid.is_empty() {
+                    uuids.insert(uuid)
+                } else {
+                    true
+                }
+            }
+            _ => true,
+        })
+        .filter_map(|v| v.get("NAME"))
+        .map(|v| v.to_owned())
+        .collect::<Vec<_>>();
```

Comment:
I suspect you could drop the turbofish here, type infererence will know it from the return value.